### PR TITLE
Support CreateNew with property values.

### LINKED ISSUE #4: Support creating immutable DTOs
We could add `CreateNew(IEnumerable<(string, object?)> properties)` to `IDtoInfo`. It would create an instance of the DTO, assigning the specified values to properties of the specified names.

This would be useful for mutable DTOs, but critical for immutable DTOs, which it could support by calling a constructor whose parameter names match read-only property names.

The introduction of immutable records with positional parameters in C# 9 makes this more pressing.

We should respect any default parameter values on the constructor.

We should support mixed mutable/immutable properties.

All property and parameter name matching should be case-insensitive.

If there are multiple possible paths, we should probably prefer constructor parameters to settable properties, for the sake of efficiency.

Adding `IDtoInfo.Constructors` could be useful, both for this behavior and for clients. Maybe detect static "constructor methods" as well?

### PR DESCRIPTION
**Pull request by ejball**
Set read-only properties via non-default constructors.

Fixes #4.

### CODE REVIEW

**Comment by ejball** [comment]
rebuild this

File: src/Faithlife.Reflection/IDtoInfo.cs [hunk 1]
```diff
@@ -33,14 +33,22 @@ public interface IDtoInfo
...
+
+		/// <summary>
+		/// Creates a new instance of the DTO, assigning properties with the specified names to the specified values.
+		/// </summary>
+		/// <param name="propertyValues">The property names and values.</param>
+		/// <remarks>If possible, the instance is created with the public default constructor, after which the specified
+		/// properties (if any) are set to the specified values. If there is no public default constructor and/or one or more
+		/// of the specified properties are read-only, the instance is created with a public constructor whose parameters
+		/// match the properties of the DTO.</remarks>
+		object CreateNew(IEnumerable<(IDtoProperty Property, object? Value)> propertyValues);
```
Comment (ddunkin) [inline src/Faithlife.Reflection/IDtoInfo.cs:46]: Nit: I'd keep the overloads next to each other.

**Review by ddunkin — APPROVED** [review]
FWIW, I wrote a test confirming it works with init-only properties as well. ae5ac6d9f1c613d008efd2e4543e9d0d17158f13

**Comment by ejball** [comment]
Good idea: [`67e5bd4` (#7)](https://github.com/Faithlife/FaithlifeReflection/pull/7/commits/67e5bd4f64e9da8f3f047339610c81ac3b15e2aa)

File: src/Faithlife.Reflection/IDtoInfo.cs (hunk 1 above)
Comment (ejball) [reply src/Faithlife.Reflection/IDtoInfo.cs:46]: Thanks, I'll change it: [`52c209b` (#7)](https://github.com/Faithlife/FaithlifeReflection/pull/7/commits/52c209b50c5fe6764c771e69fb68839aea49ca66)

Old fears of reordering interface methods...

**Pull request merged**